src/plugin_system/plugin_loader.py

```
@@ -190,49 +199,81 @@ def install_dependencies(
+
+        The plugin code keeps working because ``from scroll_display import X``
+        binds ``X`` to the class *object*, not to the sys.modules entry.
 
         Args:
-            plugin_dir: The directory of the plugin being loaded
-            plugin_id: The plugin identifier
+            plugin_id: Plugin identifier
+            plugin_dir: Plugin directory path
+            before_keys: Snapshot of sys.modules keys taken *before* exec_module
         """
         plugin_dir_str = str(plugin_dir)
```

Ruff (0.15.0)

[error] Local variable `plugin_dir_str` is assigned to but never used

Remove assignment to unused variable `plugin_dir_str`

(F841)